cmd/ci-operator-prowgen/main.go

```diff
@@ -595,11 +599,57 @@ func generateJobs(
 	}
 }
 
+// Prowgen holds the information of the prowgen's configuration file.
+type Config struct {
+	Private bool `json:"private,omitempty"`
+}
+
+type prowgenInfo struct {
+	config.Info
+	config Config
+}
+
+func readProwgenConfig(path string) *Config {
+	var pConfig *Config
+	b, err := ioutil.ReadFile(path)
+	if err != nil && !os.IsNotExist(err) {
+		return nil // , fmt.Errorf("prowgen config found in path %s but couldn't read the file: %v", path, err)
+	}
+
+	if err == nil {
+		if err := yaml.Unmarshal(b, &pConfig); err != nil {
+			return nil // , fmt.Errorf("prowgen config found in path %sbut couldn't unmarshal it: %v", path, err)
+		}
+	}
+
+	return pConfig // , nil
+}
+
 // generateJobsToDir returns a callback that knows how to generate prow job configuration
 // into the dir provided by consuming ci-operator configuration
 func generateJobsToDir(dir string, label jc.ProwgenLabel) func(configSpec *cioperatorapi.ReleaseBuildConfiguration, info *config.Info) error {
+	cache := map[string]*Config{}
```

Comment:
This can live in the `prowgenInfo`

Reply:
I don't understand. The cache is only needed in the returned function (a closure), to build a specific `prowgenInfo`. Having a pointer to the cache in `prowgenInfo` is possible, but IMHO is not needed for anything.

Reply:
I don't understand. The cache is only needed in the returned function (a closure), to build a specific prowgenInfo. Having a pointer to the cache in prowgenInfo is possible, but IMHO is not needed for anything.

Reply:
Yes, you are right. Since we are doing everything serial here, caching here makes sense.